app/src/main/AndroidManifest.xml

```diff
@@ -10,6 +14,15 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
+        <provider
+                android:name="android.support.v4.content.FileProvider"
+                android:authorities="${applicationId}.provider"
+                android:exported="false"
+                android:grantUriPermissions="true">
+            <meta-data
+                    android:name="android.support.FILE_PROVIDER_PATHS"
+                    android:resource="@xml/provider_paths"/>
```

Comment:
I pulled down your newest changes and Android Studio is showing the `Cannot find symbol @xml/provider_paths` error which is preventing me from building the app. Did you create a <paths> element with the name `provider_paths`?

I'm not familiar with FileProvider but the `Specifying Available Files` in the following link is leading me to believe that we need to create the `provider_paths.xml` file. 
https://developer.android.com/reference/android/support/v4/content/FileProvider.html